### PR TITLE
[WIP] Spike Upgrade to v6

### PR DESCRIPTION
**Pull request by danielmarbach**
Connects to #623

## Summary

An upgrade to Version 6 is possible when those issues have been taken into account

* [Redesign FailedAuditImport and FailedErrorImport](https://github.com/Particular/ServiceControl/issues/6330)

### CODE REVIEW

**Comment by danielmarbach** [comment]
Currently the biggest blockers are

* Direct usage of `TransportMessage` on the storage model, i.e. `FailedErrorImport`, so we need to copy paste the old TransportMessage definition including all related data types to ServiceControl if we don't want to patch the indexes.
* Usage of IAdvancedSatellites. We can workround that by using the satellite pipeline but we have to decide how we do the error handlling and whether we do need FLR for the satellite pipelines. We used a few custom failure managers to store failed messages inside Raven. This needs to be done directly in the behaviors registered in the corresponding satellite pipeline.

**Comment by danielmarbach** [comment]
@Particular/servicecontrol-maintainers @mat-mcloughlin continues working tomorrow on the upgrade spike and pings you in his morning. Feel free to take the PR over in the meantime

**Comment by mat-mcloughlin** [comment]
Just working on the `FeatureStartupTask`s. I'm about halfway through them.

**Comment by johnsimons** [comment]
So I was looking at converting the existing `IAdvancedSatellite` to use the new `Feature.AddSatellitePipeline` but it seems that api only caters for subqueues.
So I was thinkning of instead we host multi endpoints in the same process, so in total we would start (maybe more later on, eg heartbeats, custom checks, ...) with `particular.servicecontrol`, `audit` and 'error', thoughts ?

**Comment by danielmarbach** [comment]
Multi hosting is the better and more future proof approach.

**Comment by johnsimons** [comment]
@mat-mcloughlin  yes just split the address on https://github.com/Particular/ServiceControl/blob/master/src/ServiceControl/Contracts/Operations/ImportFailedMessage.cs#L14

**Comment by mat-mcloughlin** [comment]
Thanks @johnsimons :+1:

**Comment by johnsimons** [comment]
@mat-mcloughlin @danielmarbach I can't see us proceeding with this till https://github.com/Particular/NServiceBus/issues/3389 is done, what are your thoughts ?

**Comment by johnsimons** [comment]
Guys, after spending a day looking at this, I'm coming to the conclusion that upgrading SC to NSB v6 is plausible.
Most challenges have already been dealt with and the ones that have not (eg the satellites/multi host) we are aware of what are the limitations (see comment above).

So I think we can put this one to rest and tick it as done.

**Comment by danielmarbach** [comment]
@johnsimons @mikeminutillo @mat-mcloughlin 

I raised 

https://github.com/Particular/ServiceControl/issues/633

as an autocome of this investigation. @johnsimons could you summarize the satellite pipeline hacks and the micro endpoint discussions in an issue as well and link it here?

I think we can close this now.

**Comment by johnsimons** [comment]
@danielmarbach the summary is that in order for us to complete the upgrade to v6, we would need for core to expose either:

1. a `AddSatellitePipeline` that accepts a queue name (it currently only accepts a qualifier, mostly used for subqueues);
1. a way to create an endpoint that does not contain all the baggage eg no behaviours, no features, ..., pretty much the same as a satelite but has an endpoint;

Obviously the preferred option would be 2 because of the advantage that we would get from being able to host these new endpoints separatelly.
An issue has been raised for 2, see https://github.com/Particular/NServiceBus/issues/3389
